### ISSUE TITLE
sqitch doesn't seem to work with remote databases

### ISSUE THREAD

**Issue by srinikandula** [theory/sqitch#311]
I have a MySQL database running on Amazon RDS. I'm able to connect to it using mysql command line and also MySQLWorkbench. But when I try to use it it fails with an error 'Access denied for user ']username]'@'[ip]' (using password: YES)'. Here is the command that I'm using 

`sqitch status db:mysql://[user]:[pwd]@[host]:[port]/[databasename]`

Is sqitch supposed to work with remote hosts?

-Srini

**Comment by theory** [comment]
Is that literally the connection URI you're using? I think you want to set the username, password, host, port, and database name.

**Comment by srinikandula** [comment]
No, I masked the connection details. I do not want to past the connection details on this thread. Is there a way to check the log for connection attempt? Thank you.

**Comment by theory** [comment]
Well your error output, `'Access denied for user ']username]'@'[ip]' (using password: YES)'.`, looked weird. I guess you pasted in the details? Do you have any non-URL-safe characters in the URI?

You can get a trace from the mysql interface from Sqitch by setting the `DBI_TRACE` environment variable.

    DBI_TRACE=5 sqitch status $URI

**Comment by srinikandula** [comment]
Sure. I masked the output to remove username and IP, sorry for the confusion. The password does have some special characters and I think that is causing the issue. I changed the password. Now it seems to be working. I'll have to figure out how to use passwords with special characters. I also found out that I need to create a database with name 'sqitch' on the mysql instance. Thank you for the help.
